### ISSUE TITLE
adding, removing  capabilties, flows values

### ISSUE THREAD

**Issue by mickelluiten** [athombv/apps-sdk-issues#246]
A lot of other developers mention this feature request on slack.
With most common day's API the device  options and values can be retrieved with an API call.
In Homey SDK these options and values are static programmed thru the capabilties JSON.
E:G:
```
{
    "type": "enum",
    "title": {
        "en": "Fan Horizontal"
    },
    "platforms": [
        "local",
        "cloud"
      ],
    "getable": true,
    "setable": true,
    "uiComponent": "picker",
    "values": [
        {
            "id": "swing",
            "title": {
                "en": "Swing"
            }
        },
        {
            "id": "stop",
            "title": {
                "en": "Stop"
            }
        }
    ]
}

```
Would be great if  e.g. the values of a capabilty or/and  flow(action, trigger, condition) can be add or remove thru coding.
THis way if a vendor make changes this does not affect the Homey App.

**Comment by WeeJeWel** [comment]
We agree! What we're currently thinking about is to make a capability option that can whitelist a subset of available values, e.g. `capabilitiesOptions.fan_horizontal.allowedValues = [ 'swing' ]`.

**Comment by mickelluiten** [comment]
Great,   but the initial values (in my case)  i'am getting with a API Call , so does this mean all possible values are needed in the JSON? Or with your suggestion is still possible to add or remove the values?

**Comment by WeeJeWel** [comment]
> Great, but the initial values (in my case) i'am getting with a API Call , so does this mean all possible values are needed in the JSON?

Yes, that's correct. So it won't entirely solve your problem I think. Are these values so dynamic you cannot add them beforehand?

> Or with your suggestion is still possible to add or remove the values?

Only remove them.

**Comment by WeeJeWel** [comment]
FYI For Flows, you can use the `autocomplete` argument.

**Comment by WeeJeWel** [comment]
Maybe we should simply allow `capabilitiesOptions.fan_horizontal.values`. I'll have a chat about this with our team!

**Comment by mickelluiten** [comment]
> Yes, that's correct. So it won't entirely solve your problem I think. Are these values so dynamic you cannot add them beforehand?

Yes and No,  For most AC models there are reasonable static,  But when a vendor developer  add or remove one we still needed to change our app code. Many time (as you probaly now)  we are not informed by these changes.

Keep me posted.  Happy to test.

**Comment by AdyRock** [comment]
I would also love this feature. For a few of my apps, the API returns the list of values that apply to a capability dependent on the actual device that is connected.
The advantage of the first solution you mentioned is the actual text can be optimised / translated for each option as sometimes the list returned is for the value that is received / sent and not the nice human readable text.
So for the second method where it is completely customisable, there might need to be a way for the user to enter the displayed text for the option. Maybe that could be done in the devices settings page but I'm not sure how.

At the moment, I have to add every possible option to the list and either warn the user when they select an invalid one or relay on the API to handle it and hopefully provide a useful error message.

**Comment by oh2th** [comment]
Same issue with device settings dropdown values list, which I can get from the device API.

**Comment by oh2th** [comment]
And now with the Sensibo App, I'm looking also at have dynamic enum values, which is returned by the Sensibo API as what modes are supported by the connected A/C.
Access to update the values would be great. Though how would translations work?
